src/utils/useAttachEvent.ts

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import * as stripeJs from '@stripe/stripe-js';
+
+export const useAttachEvent = <A extends unknown[]>(
```

Comment:
I like this! This solves the on/off perf concerns and the ability to warn if no event is attached. Can you add a comment that describes the behavior of this and why it's beneficial for future versions of ourselves?